app/api/views/users.py

```diff
@@ -185,20 +188,26 @@ def signup():
     if '<' in email_address or '>' in email_address or email_address.count('@') > 1:
         return jsonify(message='Invalid email address format'), 400
 
+    if user_type == 'seller' or user_type == 'applicant':
+        if not abn:
+            return jsonify(message='You must provide an ABN value'), 400
+        # allow a value of N/A, otherwise remove any non numeric characters from the ABN and ensure it is unique
+        if not abn == 'N/A':
+            abn = re.sub(r'[^\d]', '', abn)
+            if supplier_business.abn_is_used(abn):
+                return jsonify(
+                    email_address=email_address,
+                    abn=abn,
+                    message='There is already a seller account with ABN %s' % (abn)
+                ), 409
+
     if user is not None:
         send_user_existing_password_reset_email(user.name, email_address)
         return jsonify(
             email_address=email_address,
             message="Email invite sent successfully"
         ), 200
 
-    if user_type == 'seller' or user_type == 'applicant':
-        if is_duplicate_user(email_address):
-            return jsonify(
-                email_address=email_address,
-                message='An account with this email domain already exists'
-            ), 409
```

Comment:
How are we preventing duplicate accounts if we remove this?

Reply:
This only checks that the signup email domain is unique - which is now being replaced by making sure ABN is unique. The check on whether the user already exists is untouched (it finishes like a successful signup but actually sends an email explaining the account already exists etc).